bundler/default/lib/saito/peer.js

```diff
@@ -190,19 +209,44 @@ class Peer {
                 await peer.doReqBlock(data.block_hash);
             }
         } else if (command === "SNDBLKHD") {
+console.log("RECEIVED BLOCK HEADER");
             let send_block_head_message = SendBlockHeadMessage.deserialize(message.message_data, this.app);
-            let block = await this.app.blockchain.getBlock(send_block_head_message.block_hash);
-            if (block) {
+	    let block_hash = Buffer.from(send_block_head_message.block_hash).toString("hex");
+            let is_block_indexed = this.app.blockchain.isBlockIndexed(block_hash);
+            if (is_block_indexed) {
                 console.info("SNDBLKHD hash already known: " + Buffer.from(send_block_head_message.block_hash).toString("hex"));
             } else {
-                let message_data = Buffer.from("OK", "utf-8");
-                await peer.sendResponse(message.message_id, message_data);
-                await peer.doReqBlock(send_block_head_message.block_hash);
+
+              const fetch = require('node-fetch');
+              try {
+                const url = `${this.peer.protocol}://${this.peer.host}:${this.peer.port}/block/${block_hash}`;
+                const res = await fetch(url);
+                if (res.ok) {
+                  const buffer = await res.buffer();
+                  let block = new saito.block(this.app);
+                  block.deserialize(buffer);
+                  console.log(`GOT BLOCK ${block.id} ${block.timestamp}`)
+		  this.app.mempool.addBlock(block);
+                } else {
+                  console.log(`Error fetching block: Status ${res.status} -- ${res.statusText}`);
```

Comment:
You could use `console.error` in this case to indicate the error

Reply:
Thanks for the suggestions throughout.

Have already implemented most of your suggestions in next fork.

Sweeping through the codebase and switching away from console.log is absolutely something needed. Right now we are still working on getting blocks and transactions sent from peer to peer, so haven't swept through to remove this stuff yet.